lib/Common/Memory/LargeHeapBlock.cpp

```diff
@@ -280,60 +286,112 @@ LargeHeapBlock::ReleasePagesShutdown(Recycler * recycler)
 #endif
 }
 
-template void LargeHeapBlock::ReleasePagesSweep<true>(Recycler * recycler);
-template void LargeHeapBlock::ReleasePagesSweep<false>(Recycler * recycler);
-
-template<bool pageheap>
 void
 LargeHeapBlock::ReleasePagesSweep(Recycler * recycler)
 {
     recycler->heapBlockMap.ClearHeapBlock(this->address, this->pageCount);
 
-    ReleasePages<pageheap>(recycler);
+    ReleasePages(recycler);
 }
 
+#ifdef RECYCLER_PAGE_HEAP
+__declspec(noinline)
+void LargeHeapBlock::VerifyPageHeapPattern()
+{
+    Assert(InPageHeapMode());
+    if (this->allocAddressEnd > this->address) // in case OOM while allocate heapBlockMap
+    {
+#if _M_X64
+        byte* addr = (byte*)this->allocAddressEnd;
```

Comment:
done

---
In reply to: [60992932](https://github.com/Microsoft/ChakraCore/pull/711#discussion_r60992932) [](ancestors = 60992932)